### ISSUE TITLE
Shows 12 errors on Clippy, but there really are none

### ISSUE THREAD

**Issue by mcarton** [ligthyear/clippy-service#28]
The badge on Clippy’s own `README.md` shows there are [12 errors](https://clippy.bashy.io/github/sha/Manishearth/rust-clippy/eba41430ce04ef64689acd1b991610076a3305db/log) but current `master` [compiles fine](https://travis-ci.org/Manishearth/rust-clippy/builds/115419578).
Looks like the nightly version is too old. Clippy depends on unstable _rustc_ API and often breaks on day `n+1` even though it was fine on day `n`. The service should always have the last nightly.
Also, Travis always execute

``` bash
$ rustc --version
$ cargo --version
```

for information. That would be nice to have in the log.

**Comment by gnunicorn** [comment]
Yeah, I was wondering whether that'd become a problem.

Chasing the nightly-dragon is a tough one. Do we have any idea for how long that'd be needed for clippy to work?

As it is set up now, clippy service should always run on nightly (which is broken, see #29 ) but I see your point for adding the version reporting (made a ticket as #30 ) as we'd still have a stale cache problem after that (see #31 ). I created a few separate tickets to tackle this.

**Comment by Manishearth** [comment]
I think the best thing to do is chase clippy publishes, and whenever a new clippy version is detected, update both clippy and the nightly.

**Comment by gnunicorn** [comment]
@Manishearth interesting idea. You know if there is any way we could make travis run on a push to your master?

**Comment by Manishearth** [comment]
You could add a post-build step in our travis file

**Comment by gnunicorn** [comment]
@Manishearth only wanted to figure out how hard that would be: turns out not that hard. Send a PR your way to trigger a rebuild of clippy-service whenever rust-clippy master is successfully build. Ping me about the access token before you want to merge it, and I'll give it to you :) .
